pootle/core/state.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class ItemState(object):
+
+    def __init__(self, state, state_type, **kwargs):
+        self.state = state
+        self.state_type = state_type
+        self.kwargs = kwargs
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, self.__class__)
+            and other.state_type == self.state_type
+            and other.state == self.state
+            and (sorted(self.kwargs.items())
+                 == sorted(other.kwargs.items())))
+
+    def __str__(self):
+        return (
+            "<%s(%s): %s %s>"
+            % (self.__class__.__name__,
+               self.state.context,
+               self.state_type,
+               self.kwargs))
+
+    def __getattr__(self, k):
+        if k in self.kwargs:
```

Comment:
it gets covered in the next PR, but yep, it should be tested in its own right - ill add